proto/redpanda/core/admin/v2/shadow_link.proto

```diff
@@ -64,6 +65,20 @@ service ShadowLinkService {
             authz: SUPERUSER
         };
     }
+
+    // RPCs for the ShadowTopic resource
+
+    rpc GetShadowTopic(GetShadowTopicRequest) returns (GetShadowTopicResponse) {
+        option (redpanda.core.pbgen.rpc) = {
+            authz: SUPERUSER
+        };
+    }
+    rpc ListShadowTopics(ListShadowTopicsRequest)
+        returns (ListShadowTopicsResponse) {
+        option (redpanda.core.pbgen.rpc) = {
+            authz: SUPERUSER
+        };
+    }
```

Comment:
I know this is just CRUD, but since it's a read only resource comments about what you are getting here, how the are created and why you would want to call these APIs is useful